website/source/docs/provisioning/file.html.md

```diff
@@ -49,3 +85,31 @@ The file provisioner takes only two options, both of which are required:
   the source will be uploaded to. The file/folder is uploaded as the SSH user
   over SCP, so this location must be writable to that user. The SSH user can be
   determined by running `vagrant ssh-config`, and defaults to "vagrant".
+
+## Caveats
+
+While the file provisioner does support trailing slashes or "globing", this can
+lead to some confusing results due to the underlying tool used to copy files and
+folders between the host and guests. For example, if you have a source and
+destination defined below:
+
+      config.vm.provision "file", source: "~/pathfolder", destination: "/remote/newlocation/"
+      config.vm.provision "file", source: "/otherfolder/.", destination: "/remote/otherlocation"
+
```

Comment:
I think that describing each of these individually will the information a bit clearer.